src/libraries/Microsoft.Extensions.Http/src/Logging/LogHelper.cs

```diff
@@ -151,9 +151,16 @@ public static void LogRequestPipelineEnd(this ILogger logger, HttpResponseMessag
                 return null;
             }
 
-            string uriString = uri.IsAbsoluteUri
-                ? uri.GetComponents(UriComponents.AbsoluteUri & ~UriComponents.UserInfo, UriFormat.UriEscaped)
-                : uri.ToString();
+            if (!uri.IsAbsoluteUri)
```

Comment:
This should be after `s_disableUriQueryRedaction` check

Reply:
Because, to my understanding, the switch should return everything to the old behavior

Reply:
I don't care that much about not being able to do a precise redaction of a relative uri, what I'm not happy about is introducing a breaking change without a straightforward way to opt-out.

But whatever. While I was able to find some interesting usecases for UserInfo in Uri in general, but for a plain HttpClientFactory case it _should_ be rare. Matching the OTel logic sounds ok-ish... (but we don't fully match their format, do we?)